src/mbgl/map/map.cpp

```diff
@@ -352,6 +352,8 @@ void Map::Impl::loadStyleJSON(const std::string& json) {
 
     // force style cascade, causing all pending transitions to complete.
     style->cascade(Clock::now(), mode);
+    
+    view.notifyMapChange(MapChangeDidFinishLoadingStyle);
```

Comment:
In that case, does it make more sense to add the signals to the Style itself? 

In https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/style/style.cpp#L95 we can best check for parse errors.